src/librustc/ty/sty.rs

```diff
@@ -1156,7 +1156,6 @@ impl<'a, 'tcx, 'gcx> PolyExistentialProjection<'tcx> {
 
 impl DebruijnIndex {
     pub fn new(depth: u32) -> DebruijnIndex {
-        assert!(depth > 0);
```

Comment:
I think `from_depth` also needs to be change to remove the shift by 1.

Reply:
Good point. I'll give it a go locally.